SignalServiceKit/src/Loki/API/LokiAPI.swift

```diff
@@ -270,6 +270,9 @@ public final class LokiAPI : NSObject {
     private static func updateLastMessageHashValueIfPossible(for target: LokiAPITarget, from rawMessages: [JSON]) {
         if let lastMessage = rawMessages.last, let hashValue = lastMessage["hash"] as? String, let expirationDate = lastMessage["expiration"] as? Int {
             setLastMessageHashValue(for: target, hashValue: hashValue, expirationDate: UInt64(expirationDate))
+            if UserDefaults.standard[.isUsingFullAPNs] {
+                LokiPushNotificationManager.acknowledgeDeliveryForMessage(with: hashValue, expiration: expirationDate, hexEncodedPublicKey: userHexEncodedPublicKey)
```

Comment:
Instead of doing this here let's add it to `LokiAPI.getMessages()` and `LokiPoller.poll(_:seal:)`. Right now it's an unexpected side effect of calling `updateLastMessageHashValueIfPossible(for:from:)` as you would expect that to just update the last message hash value.